File: leetcode_practice/leetcode_977.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int left = 0;
        int right =nums.size()-1;
        int index = nums.size()-1;
        vector<int>ans(nums.size());
        while(left<=right){
            long long leftsq =(long long) nums[left]*nums[left];
            long long rightsq = (long long) nums[right]*nums[right];
            if(leftsq > rightsq){
                ans[index] = leftsq;
                left++;
            }else{
                ans[index]= rightsq;
                right--;
            }
            index--;
        }
        return ans;
    }
};

File: leetcode_py/leetcode_977.py
class Solution(object):
    def sortedSquares(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        left = 0
        n = len(nums)
        right = n - 1
        index = n - 1
        ans = [0] * n
        while left <= right:
            leftsq = nums[left] * nums[left]
            rightsq = nums[right] * nums[right]
            if leftsq > rightsq:
                ans[index] = leftsq
                left += 1
            else:
                ans[index] = rightsq
                right -= 1
            index -= 1
        return ans
